Content.Server/GameObjects/Components/Damage/DamageOnToolInteractComponent.cs

```diff
@@ -0,0 +1,108 @@
+            serializer.DataField(ref UsingMultitool,    "usingMultitool",   false);
+            serializer.DataField(ref UsingPrying,       "usingPrying",      false);
+            serializer.DataField(ref UsingScrewing,     "usingScrewing",    false);
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _actSystem = _entitySystemManager.GetEntitySystem<ActSystem>();
+        }
+
+            public bool InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+
+            //Used to support multiple tool interactions.
+            //If two tools were supported then the first check would return false and be past straight to return breaking function and second check would never be carried out.
+            //assumes interaction can only be with one tool at one time.
+            bool result = false; 
+
+            if (UsingWelding)
+            {
+                if (eventArgs.Using.HasComponent<WelderComponent>())
+                {
+                    WelderComponent tool = eventArgs.Using.GetComponent<WelderComponent>();
+                    if(tool.WelderLit)
+                    if (tool.HasQuality(ToolQuality.Welding))
+                    result = CallDamage(eventArgs);
+                }
+                if (result is true) return true;
+            }
+
+            if (UsingAnchoring) { result = HandleToolInteract(eventArgs, ToolQuality.Anchoring); if (result is true) return true; }
+
+            if (UsingCutting) { result = HandleToolInteract(eventArgs, ToolQuality.Cutting); if (result is true) return true; }
+
+            if (UsingMultitool) { result = HandleToolInteract(eventArgs, ToolQuality.Multitool); if (result is true) return true; }
+
+            if (UsingPrying) { result = HandleToolInteract(eventArgs, ToolQuality.Prying); if (result is true) return true; }
+
+            if (UsingScrewing) { result = HandleToolInteract(eventArgs, ToolQuality.Screwing); if (result is true) return true; }
+
+            return result;
+        }
+
+        protected bool HandleToolInteract(InteractUsingEventArgs eventArgs, ToolQuality quality)
+        {
+        if (eventArgs.Using.HasComponent<ToolComponent>())
+        {
+            if (eventArgs.Using.GetComponent<ToolComponent>().HasQuality(quality)) return CallDamage(eventArgs);
+        }
+            return false;
+        }
+
+        protected bool CallDamage(InteractUsingEventArgs eventArgs)
+        {
+
+            if (eventArgs.Target.HasComponent<DamageableComponent>())
+            {
+                DamageableComponent damageable = eventArgs.Target.GetComponent<DamageableComponent>();
+                damageable.TakeDamage(Shared.GameObjects.DamageType.Heat, Damage, eventArgs.Using, eventArgs.User);
+                return true;
+            }
+            else if (Damage > 0)
+            {
+                _actSystem.HandleDestruction(Owner, true);
+                return true;
+            }
```

Comment:
I'm not sure you should be doing this directly. Did you find that this code was necessary?

Reply:
Not really. Its only a fall back implementation if no damageable component is in place. I don't want to force people to use damageable component on everything though so thats the solution I came up with. If its dangerous or there is a better method I’ll gladly change it.

Reply:
It's a nice idea, but given that this is a `Damage`-ing component, I think it makes more sense to only have an effect when dealing damage is actually possible.

Reply:
You're right. People should assume this component requires a damagable component. Also I have thought about it some more. If someone was expecting for an entity to only take half damage to say change the sprite then this could really throw them as it won't give an error. So it’s actually an undesired behaviour.

 How would I change it to a log Warning message sent to the server console instead?

Reply:
You *could* use `Logger.WarningS("damageOnTool", ...)` to add the warning you want. However, failing fast is generally preferred in this codebase - so I would recommend you not use `HasComponent` (or `TryGetComponent`) and instead just use `GetComponent` directly. It will throw an exception if the component is missing, so the error will be easier to spot.

You can additionally put `Owner.EnsureComponent<DamageableComponent>()` in `Initialize` so that, if the damageable component is missing, the error will be raised when the entity is first created - but this isn't a requirement.